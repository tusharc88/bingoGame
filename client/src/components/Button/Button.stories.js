import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';
import Button from './index';

storiesOf('Button', module)
  .addDecorator(withKnobs)
  .addDecorator(centered)
  .addWithInfo('default', <Button />, () => (
    <Button onClick={action('onClick')} disabled={boolean('disabled', false)}>
      {text('children', 'Click Me!')}
    </Button>
  ));
