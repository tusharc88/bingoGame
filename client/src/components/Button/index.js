import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ButtonWrapper = styled.div`
  height: 40px;
  width: 100px;
  font-size: 15px;
  border: 1px solid #989a9b;
  border-radius: 5px;
  background-color: #cdd2d4;
  color: #FFF;
  text-align: center;
  vertical-align: middle;
  line-height: 35px;
  cursor: ${props => (!props.disabled ? 'pointer' : 'default')}
  transition: all 0.8s ease-in-out;
  transition-property: background-color box-shadow;
  ${props =>
    !props.disabled &&
    `&:hover {
    background-color: #acb0b3;
    box-shadow: 0px 15px 10px -10px #373839;
    transform: translateY(3px);
    transition: all 0.3s ease-in-out;
    transition-property: background-color box-shadow;
  }`};
`;

const Button = ({ children, onClick, disabled }) => {
  const handleClick = () => {
    if (!disabled) onClick();
  };
  return (
    <ButtonWrapper onClick={handleClick} disabled={disabled}>
      {children}
    </ButtonWrapper>
  );
};

Button.propTypes = {
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
};

Button.defaultProps = {
  onClick: () => {},
  disabled: false,
};

export default Button;
