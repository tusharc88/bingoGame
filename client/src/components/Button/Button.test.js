import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Button from './index';

test('Renders with default props', () => {
  const tree = renderer.create(<Button>Click Me</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders with onClick prop', () => {
  const tree = renderer
    .create(<Button onClick={() => {}}>Click Me</Button>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
