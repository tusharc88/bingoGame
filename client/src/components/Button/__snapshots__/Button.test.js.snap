// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with default props 1`] = `
.c0 {
  height: 40px;
  width: 100px;
  font-size: 15px;
  border: 1px solid #989a9b;
  border-radius: 5px;
  background-color: #cdd2d4;
  text-align: center;
  vertical-align: middle;
  line-height: 35px;
  cursor: pointer;
  -webkit-transition: all 0.8s ease-in-out;
  transition: all 0.8s ease-in-out;
  -webkit-transition-property: background-color box-shadow;
  transition-property: background-color box-shadow;
}

.c0:hover {
  background-color: #acb0b3;
  box-shadow: 0px 15px 10px -10px #373839;
  -webkit-transform: translateY(3px);
  -ms-transform: translateY(3px);
  transform: translateY(3px);
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  -webkit-transition-property: background-color box-shadow;
  transition-property: background-color box-shadow;
}

<div
  className="c0"
  disabled={false}
  onClick={[Function]}
>
  Click Me
</div>
`;

exports[`Renders with onClick prop 1`] = `
.c0 {
  height: 40px;
  width: 100px;
  font-size: 15px;
  border: 1px solid #989a9b;
  border-radius: 5px;
  background-color: #cdd2d4;
  text-align: center;
  vertical-align: middle;
  line-height: 35px;
  cursor: pointer;
  -webkit-transition: all 0.8s ease-in-out;
  transition: all 0.8s ease-in-out;
  -webkit-transition-property: background-color box-shadow;
  transition-property: background-color box-shadow;
}

.c0:hover {
  background-color: #acb0b3;
  box-shadow: 0px 15px 10px -10px #373839;
  -webkit-transform: translateY(3px);
  -ms-transform: translateY(3px);
  transform: translateY(3px);
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  -webkit-transition-property: background-color box-shadow;
  transition-property: background-color box-shadow;
}

<div
  className="c0"
  disabled={false}
  onClick={[Function]}
>
  Click Me
</div>
`;
